src/hooks/helper/claim.ts

```diff
@@ -0,0 +1,287 @@
+import { BatchTxs } from './index';
+import { getAddressEnum } from './../../store/dapp-staking/calculation';
+import { Struct, Option } from '@polkadot/types';
+import { ApiPromise } from '@polkadot/api';
+
+interface ContractEraStake extends Struct {
+  readonly contractRewardClaimed: boolean;
+  readonly numberOfStakers: string;
+  readonly total: string;
+}
+
+export interface StakersInfo extends Struct {
+  // Todo: fix type annotation
+  readonly stakes: any[];
+}
+
+const getStakerInfo = async ({
+  dappAddress,
+  api,
+  senderAddress,
+  currentEra,
+}: {
+  dappAddress: string;
+  senderAddress: string;
+  api: ApiPromise;
+  currentEra: number;
+}): Promise<{ firstEraToClaimForStaker: number; numberOfUnclaimedEra: number } | null> => {
+  const data = await api.query.dappsStaking.stakersInfo<StakersInfo>(
+    senderAddress,
+    getAddressEnum(dappAddress)
+  );
+  if (data && !data.isEmpty) {
+    const stakes = data.stakes;
+    const firstEraToClaimForStaker = Number(stakes[0].toHuman().era);
+    const numberOfUnclaimedEra = currentEra - firstEraToClaimForStaker;
+
+    return { firstEraToClaimForStaker, numberOfUnclaimedEra };
```

Comment:
Unfortunately, it's a bit more complicated than this :)

You have to account for possible _holes_ between eras.
Please check [this](https://github.com/AstarNetwork/astar-frame/pull/6/files#diff-74f8acb4bf207fcaa637a96dcba0e2d302253ac759392b2799bf2f5a32938745R161-R187).

Basically, you have to account for eras in which stake was 0.

E.g.:
current_era = 10;
 `[<1, 1000>, <2, 1500>, <3, 0>, <7, 1300>]` means that you can claim eras:
**1, 2, 7, 8, 9**

Reply:
Thank you, @Dinonard.
I fixed this issue in [this](https://github.com/AstarNetwork/astar-apps/pull/258/commits/77fbfd9ac61aef0a6a1645f9aa342e6b84d1cdb9) commit.

Demo: https://www.loom.com/share/d6fcde392cd94c14b3951cc84d39398b

![Screenshot 2022-02-28 at 1 06 30 PM](https://user-images.githubusercontent.com/92044428/155936503-43a9ab10-9732-47e4-9136-0aaa95f4f34f.png)

![Screenshot 2022-02-28 at 1 09 12 PM](https://user-images.githubusercontent.com/92044428/155936510-96af03e1-b980-41bc-865d-50c2694d7c8b.png)